azure/activity_logs_monitoring/index.js

```diff
@@ -381,14 +381,38 @@ class EventhubLogHandler {
                     this.records.push(originalRecord);
                 }
             } else {
-                this.records.push(originalRecord);
+                this.records.push(this.fixPropertiesJsonString(originalRecord));
             }
         } else {
             record = this.addTagsToStringLog(record);
             this.records.push(record);
         }
     }
 
+    fixPropertiesJsonString(record) {
+        try {
+            // Check if properties field is a string
+            if (typeof record.properties === 'string') {
+                // If it is a string, check if it is a malformed JSON String
+                // By checking for ':'
+                if (record.properties.includes("':'")) {
+                    // If the check is true, find / replace single quotes
+                    // with double quotes, to make a proper JSON
+                    // which is then converted into a JSON Object
+                    record.properties = JSON.parse(record.properties.replace(/'/g, '"'));
+                    return record;
+                }else {
+                    return record;
+                }
+            } else {
+                return record;
+            }
+        } catch {
+            this.context.error('Unable to fix properties field to JSON Object');
+            return record;
+        }
+    }
```

Comment:
```suggestion
    fixPropertiesJsonString(record) {
        // Check if properties field is a malformed JSON String
        if (Object.hasOwn(record, 'properties') && (typeof record.properties === 'string') && (record.properties.includes("':'"))) {
            try {
                // If the check is true, find and replace single quotes
                // with double quotes, to make a proper JSON
                // which is then converted into a JSON Object
                parsedProperties = JSON.parse(record.properties.replace(/'/g, '"'));
                record.properties = parsedProperties;
            } catch {
                this.context.error('Unable to fix properties field to JSON Object');
            }
        return record;
    }
```
This function has a lot of return paths that all seem to do the same thing. This could be simplified a bit to have a single return and combine our checks into a single statement to verify whether or not record.properties has one of these malformed objects.

We'll need to do a little verification on our end to make this work with live data and double check the behavior of nested message and level fields.

Reply:
@mattsp1290 I went ahead and took the changes that you've recommended. I added a couple of return clauses that ensures:
- When the if statement fails, we should return the record as is
- If the conversion fails, we should return the record as is

```js
    fixPropertiesJsonString(record) {
        // Check if properties field is a malformed JSON String
        if (Object.hasOwn(record, 'properties') && (typeof record.properties === 'string') && (record.properties.includes("':'"))) {
            try {
                // If the check is true, find and replace single quotes
                // with double quotes, to make a proper JSON
                // which is then converted into a JSON Object
                record.properties = JSON.parse(record.properties.replace(/'/g, '"'));
                return record;
            } catch {
                this.context.error('Unable to fix properties field to JSON Object');
                return record;
            }
        } else {
            return record;
        }
    }
```

I have also added a couple of additional test cases, that will test nested objects.